tests/test_cases/test_cocotb/test_logging.py

```diff
@@ -39,3 +43,43 @@ def test_logging_with_args(dut):
     dut._log.warning("Testing multiple line\nmessage")
 
     yield Timer(100)  # Make it do something with time
+
+
+@cocotb.test()
+def test_logging_default_config(dut):
+    # The cocotb.log module is shadowed by an instance of
+    # cocotb.log.SimBaseLog()
+    from cocotb.log import default_config as log_default_config
+
+    cocotb_log = logging.getLogger('cocotb')
+
+    # Save pre-test configuration
+    log_level_prev = cocotb_log.level
+    os_environ_prev = os.environ.copy()
+
+    try:
+        # Set a valid log level
+        os.environ['COCOTB_LOG_LEVEL'] = 'DEBUG'
+        log_default_config()
+        assert cocotb_log.level == logging.DEBUG, cocotb_log.level
+
+        # Try to set log level to an invalid log level
+        os.environ['COCOTB_LOG_LEVEL'] = 'INVALID_LOG_LEVEL'
+        try:
+            log_default_config()
+        except ValueError:
+            pass
+        else:
+            assert 0, "Expected ValueError wasn't raised."
+
+        # Try to set log level to a valid log level with wrong capitalization
+        os.environ['COCOTB_LOG_LEVEL'] = 'error'
+        log_default_config()
+        assert cocotb_log.level == logging.ERROR, cocotb_log.level
+
+    finally:
+        # Restore pre-test configuration
+        os.environ = os_environ_prev
+        cocotb_log.level = log_level_prev
+
+    yield Timer(1)
```

Comment:
```suggestion
    yield Timer(1, 'ns')
```
and maybe even use ``await``?

Reply:
Generator-based coroutines are the norm in this file, I'd keep things consistent for the time being (until we convert the file as a whole). The same goes for a Timer without any unit; it's only doing something, the actual trigger doesn't matter at all.